scalafix-tests/src/test/resources/checkSyntax/NoUnusedImports.source

```diff
@@ -0,0 +1,44 @@
+rewrites = [NoUnusedImports]
+<<< remove unused imports
```

Comment:
There is a test suite here https://github.com/scalacenter/scalafix/blob/996f30ff87ad9442c53a06ccf505ac6e845b9672/scalafix-tests/src/test/resources/checkSyntax/OrganizeImportsRemoveUnused.source which might be relevant to take a look at. It contains a few tricky cases like macro/language imports which the compiler message approach should handle.